lib/theme_mode_handler.dart

```diff
@@ -43,29 +50,21 @@ class ThemeModeHandlerState extends State<ThemeModeHandler> {
   @override
   void initState() {
     super.initState();
-    _loadThemeMode().then((ThemeMode value) {
+    _loadThemeMode().then((value) {
       setState(() {
         _themeMode = value;
       });
     });
   }
 
-  Future<void> setThemeMode(ThemeMode value) async {
-    setState(() {
-      _themeMode = value;
-    });
-
-    final SharedPreferences prefs = await SharedPreferences.getInstance();
-
-    await prefs.setString(
-      widget.sharedPreferencesKey,
-      value.toString(),
-    );
+  /// Updates the themeMode and calls `manager.saveThemeMode`.
+  Future<void> saveThemeMode(ThemeMode value) async {
+    setState(() => _themeMode = value);
```

Comment:
Extract the setState to a function to keep single responsability, you also can reuse the same function to initState and here.